test/broadcast.jl

```diff
@@ -419,34 +418,67 @@ end
 @test let z = 1; A = broadcast!(() -> z += 1, zeros(2)); A[1] != A[2]; end
 @test let z = 1; A = broadcast!(x -> z += x, zeros(2), 1); A[1] != A[2]; end
 
-# broadcasting for custom AbstractArray
-struct Array19745{T,N} <: AbstractArray{T,N}
+## broadcasting for custom AbstractArray
+abstract type ArrayData{T,N} <: AbstractArray{T,N} end
+Base.getindex(A::ArrayData, i::Integer...) = A.data[i...]
+Base.setindex!(A::ArrayData, v::Any, i::Integer...) = setindex!(A.data, v, i...)
+Base.size(A::ArrayData) = size(A.data)
+Base.broadcast_similar(f, ::Broadcast.ArrayStyle{A}, ::Type{T}, inds::Tuple, As...) where {A,T} =
+    A(Array{T}(length.(inds)))
```

Comment:
`Array{T}(length.(inds))` -> `Array{T}(uninitialized, length.(inds))` in anticipation of conflict? :)